src/account/AccountStorage.sol

```diff
@@ -41,17 +41,19 @@ struct SelectorData {
     bool isPublic;
     // Whether or not a shared validation function may be used to validate this function.
     bool allowSharedValidation;
-    // How many times a `PRE_HOOK_ALWAYS_DENY` has been added for this function.
-    // Since that is the only type of hook that may overlap, we can use this to track the number of times it has
-    // been applied, and whether or not the deny should apply. The size `uint48` was chosen somewhat arbitrarily,
-    // but it packs alongside `plugin` while still leaving some other space in the slot for future packing.
-    uint48 denyExecutionCount;
-    // User operation validation and runtime validation share a function reference.
+    // The execution hooks for this function selector.
+    EnumerableSet.Bytes32Set executionHooks;
+    // Which validation functions are associated with this function selector.
     EnumerableSet.Bytes32Set validations;
+}
+
+struct ValidationData {
+    // Whether or not this validation can be used as a shared validation function.
+    bool isShared;
+    // Whether or not this validation is a signature validator.
+    bool isSignatureValidation;
```

Comment:
is it possible to add `defaultValidations` as a bool field here?

Reply:
Yep that's what the `isShared` flag is doing, it's whether or not it counts as a "default" validation. I think the word "default" implies that there's only one ("the default"), but in reality there can be many validations in the shared "default" pool, so I was kinda trying to avoid using that term. But I totally see how this is confusing, can change it or add a comment if it would help.

Reply:
Would be helpful yeah! How about global validators? Global validators are able to perform validation across all native selectors + installed execution selectors that opt in to global validation

Reply:
That's the same thing, right?

Reply:
yup, just a name suggestion, my contribution it to the list of names for the naming meeting :p

Reply:
We can punt, but I do like the idea of using "global" here. I think it's fine that "global validation functions" don't necessarily apply to all selectors, just like how global variables can be overriden.

Reply:
> We can punt, but I do like the idea of using "global" here. I think it's fine that "global validation functions" don't necessarily apply to all selectors, just like how global variables can be overriden.

yeah that's the behavior I expect as well. We often adhere to `default vs overridden` in our codebase, but the same idea is the same.